### PR TITLE
add djones filter

### PR DESCRIPTION
**Pull request by DylanJones**

### CODE REVIEW

File: cslbot/helpers/textutils.py [hunk 1]
```diff
@@ -368,12 +368,34 @@ def gen_sanitize(msg):
 def gen_intensify(msg):
     return "[%s INTENSIFIES]" % msg.upper()
 
+def gen_djones(msg):
+    keyboard = [
+            'qwertyuiop',
+            '\0asdfghjkl',
+            '\0\0zxcvbnm\0\0',
```
Comment (pefoley2) [inline cslbot/helpers/textutils.py]: Why embedded nulls?

File: cslbot/helpers/textutils.py [hunk 2]
```diff
@@ -368,12 +368,34 @@ def gen_sanitize(msg):
...
+            ]
+    outmsg = ''
+    for letter in msg:
+        if random() < 0.15 and letter in ''.join(keyboard):
+            r = (0 if letter in keyboard[0] else (1 if letter in keyboard[1] else 2))
```
Comment (pefoley2) [inline cslbot/helpers/textutils.py]: These variable names are a bit short, this isn't go :P

File: cslbot/helpers/textutils.py [hunk 3]
```diff
@@ -368,12 +368,34 @@ def gen_sanitize(msg):
...
+            c = keyboard[r].find(letter)
+            nr = r + randint(-1, 1)
+            nc = c + randint(-1, 1)
+            while nr < 0 or nr >= len(keyboard) or nc < 0 or nc >= len(keyboard[0]) or keyboard[nr][nc] == '\0':
```
Comment (pefoley2) [inline cslbot/helpers/textutils.py]: This conditional is a bit much, why not:
while True:
if ...: break
if ...: break

File: cslbot/helpers/textutils.py (hunk 1 above)
Comment (DylanJones) [reply cslbot/helpers/textutils.py]: Nulls make the array rectangular so vertical offsets make sense

File: cslbot/helpers/textutils.py (hunk 2 above)
Comment (DylanJones) [reply cslbot/helpers/textutils.py]: just nr, nc? or others?

**Comment by pefoley2** [comment]
Build is still broken.

File: cslbot/helpers/textutils.py (hunk 2 above)
Comment (pefoley2) [reply cslbot/helpers/textutils.py]: all the 1/2 char variables, honestly

File: cslbot/helpers/textutils.py (hunk 3 above)
Comment (DylanJones) [reply cslbot/helpers/textutils.py]: is it really necesary to break this up?  it kinda feels wrong to have a `while True` with break statements, and it fits within the character limit

**Comment by pefoley2** [comment]
You're really overdoing it on the type annotations, FYI.

File: cslbot/helpers/textutils.py [hunk 4]
```diff
@@ -369,11 +369,50 @@ def gen_intensify(msg):
     return "[%s INTENSIFIES]" % msg.upper()
 
 
+def gen_djones(msg: str) -> str:
```
Comment (pefoley2) [inline cslbot/helpers/textutils.py]: https://google.github.io/styleguide/pyguide.html?showone=Comments#Comments

"On the other hand, never describe the code. Assume the person reading the code knows Python (though not what you're trying to do) better than you do."

You have way the hell too many comments here, and the only type annotations that actually add value are the ones in the method signature.

**Pull request merged**